refactor(ui-components): Set up ui-components library with ProgressEllipsis

carolinemodic opened the pull request:
Sets up a new shared library for ui components, with a basic initial component of ProgressEllipsis. I chose this one as it doesn't import any other components and has not diverged across our different repos. 

@eventualbuddha I'm not entirely sure I set up the linting properly... I couldn't get stylelint:fix to work properly, but was having issues with it in other repos as well so I just removed it. The rest of the linting setup is based on election-manager.

beausmith commented on libs/ui-components/.gitignore:
```diff
@@ -0,0 +1 @@
+/lib
```
Ignore the `/coverage` dir:

```suggestion
/lib

# testing
/coverage
```

beausmith commented:
Request: change from `ui-components` to `ui` so that the name is more general to the theme of user interface which could include other UI-related files: images, fonts, themes, etc.

beausmith reviewed (approved):
Sorry, I had intended to approve PR with the requested changes.

carolinemodic commented:
Renamed to /ui
Fixed missing .gitignore things
Fixed tsconfigs in the importing apps not including @votingworks/ui, once I did this making changes to a component in /ui automatically triggers a re-compiling while running an app (i.e. election-manager) and the changes autoload.

eventualbuddha commented on apps/bmd/package.json:
```diff
@@ -89,6 +89,7 @@
     "@types/react-router-dom": "^5.1.5",
     "@types/styled-components": "^5.0.1",
     "@votingworks/ballot-encoder": "workspace:*",
+    "@votingworks/ui": "workspace:*",
```
This should sort after `@votingworks/qrcode.react`, right?

eventualbuddha commented on libs/ui/.gitignore:
```diff
@@ -0,0 +1,23 @@
+.env.local
+.env.development.local
+.env.test.local
+.enc.production.local
```
```suggestion
.env.production.local
```

eventualbuddha commented on libs/ui/package.json:
```diff
@@ -0,0 +1,79 @@
+  "private": true,
+  "description": "Generic UI components for VotingWorks UIs",
+  "license": "GPL-3.0",
+  "author": "Caroline Modic",
```
I probably have one with my name on it, but I intend to change any to this:
```suggestion
  "author": "VotingWorks <eng@voting.works>",
```

eventualbuddha commented on libs/ui/package.json:
```diff
@@ -0,0 +1,79 @@
+    ]
+  },
+  "dependencies": {
+    "@testing-library/react": "^11.2.6",
+    "@types/styled-components": "^5.1.9",
+    "prettier": "^2.1.1",
```
These belong in `devDependencies`, right?
